### PR TITLE
Add phonetic token filter docs #8242

### PR DESCRIPTION
**Pull request by AntonEliatra**
### Description
a phonetic token filter docs

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/8242

### Version
all

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by vagimeli** [comment]
@udabhas Will you review this PR for technical accuracy, or have a peer review it? Thank you.

File: _analyzers/token-filters/phonetic.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
+---
+layout: default
+title: Phonetic
+parent: Token filters
+nav_order: 330
+---
+
+# Phonetic token filter
+
+The `phonetic` token filter transforms tokens into their phonetic representations, enabling more flexible matching of words that sound similar but are spelled differently. This is particularly useful for searching names, brands, or other entities which users might spell differently but pronounce similarly.
```
Comment (natebower) [inline _analyzers/token-filters/phonetic.md:10]:
```suggestion
The `phonetic` token filter transforms tokens into their phonetic representations, enabling more flexible matching of words that sound similar but are spelled differently. This is particularly useful for searching names, brands, or other entities that users might spell differently but pronounce similarly.
```

File: _analyzers/token-filters/phonetic.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+The `phonetic` token filter is not included in OpenSearch distributions by default. To use this token filter, you must first install the `analysis-phonetic` plugin as follows and then restart OpenSearch:
+
+```bash
+./bin/opensearch-plugin install analysis-phonetic
+```
+{% include copy.html %}
+
+For more information about installing plugins, see [Installing plugins]({{site.url}}{{site.baseurl}}/install-and-configure/plugins/)
```
Comment (natebower) [inline _analyzers/token-filters/phonetic.md:19]:
```suggestion
For more information about installing plugins, see [Installing plugins]({{site.url}}{{site.baseurl}}/install-and-configure/plugins/).
```

File: _analyzers/token-filters/phonetic.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+{: .note}
+
+## Parameters
+
+The `phonetic` token filter can be configured with the following parameters.
+
+Parameter | Required/Optional | Data type | Description
+:--- | :--- | :--- | :--- 
+`encoder` | Optional | String | Specifies the phonetic algorithm to use.<br><br>Valid values are:<br>- `metaphone` (default)<br>- `double_metaphone`<br>- `soundex`<br>- `refined_soundex`<br>- `caverphone1`<br>- `caverphone2`<br>- `cologne`<br>- `nysiis`<br>- `koelnerphonetik`<br>- `haasephonetik`<br>- `beider_morse`<br>- `daitch_mokotoff ` 
+`replace` | Optional | Boolean | Whether to eplace the original token. If `false`, the original token is included in the output along with the phonetic encoding. Default is `true`.
```
Comment (natebower) [inline _analyzers/token-filters/phonetic.md:29]:
```suggestion
`replace` | Optional | Boolean | Whether to replace the original token. If `false`, the original token is included in the output along with the phonetic encoding. Default is `true`.
```

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Just a few changes. Thanks!

**Review by kolchfa-aws — APPROVED** [review]

**Pull request merged**